core/tx_pool.go

```diff
@@ -836,6 +839,11 @@ func (pool *TxPool) addTx(tx *types.Transaction, local bool) error {
 
 // addTxs attempts to queue a batch of transactions if they are valid.
 func (pool *TxPool) addTxs(txs []*types.Transaction, local bool) []error {
+	// caching signature check result before holding lock significantly improve concurrency
```

Comment:
// Cache senders in transactions before obtaining lock (pool.signer is immutable)